cmds/boot/boot.go

```diff
@@ -0,0 +1,451 @@
+		return err
+	}
+	if sig != bootableMBR {
+		err := errors.New("Not a bootable device")
+		return err
+	}
+	return nil	
+}
+
+// getDevicePartList returns all devices attached to a specific name like /dev/sdaX where X can move from 0 to 127
+// FIXME no support for devices which are included into subdirectory within /dev
+func getDevicePartList(path string) ([]string,error) {
+	var returnValue []string
+	files, err := ioutil.ReadDir("/dev/")
+	if err != nil {
+		return returnValue, err
+	}
+	for _, file := range files {
+		if strings.HasPrefix(file.Name(), path) {
+			// We shall not return full device name
+			if file.Name() != path {
+				// We shall check that the remaining part is a number
+				returnValue = append(returnValue, file.Name())
+			}
+		}
+	}
+	return returnValue, nil
+}
+
+// getSupportedFilesystem returns all block file system supported by the linuxboot kernel
+func getSupportedFilesystem() ([]string, error) {
+	var returnValue []string
+	var err error
+	fs, err := ioutil.ReadFile("/proc/filesystems") 
+	if err != nil {
+		return returnValue, err
+	}
+	for _, f := range strings.Split(string(fs), "\n") {
+		n := strings.Fields(f)
+		if len(n) != 1 {
+			continue
+		}
+		returnValue=append(returnValue,n[0])
+	}
+	return returnValue, err
+
+}
+
+// mountEntry tries to mount a specific block device
+func mountEntry(path string, supportedFilesystem []string) (bool,error) {
+	var returnValue bool
+	var err error
+	exist, err := os.Stat("/u-root")
+        if exist == nil {
+                err=syscall.Mkdir("/u-root", 0777)
+                if ( err != nil ) {
+                        return false,err
+                }
+        }
+	var flags uintptr
+	// Was supposed to be unecessary for kernel 4.x.x
+	if verbose {
+		log.Printf("/dev/" + path)
+	}
+	for _, filesystem := range supportedFilesystem {
+		flags = syscall.MS_MGC_VAL
+		// Need to load the filesystem kind supported
+		exist, err = os.Stat("/u-root/"+path)
+                if exist == nil {
+                        err = syscall.Mkdir("/u-root/"+path, 0777)
+                        if ( err != nil ) {
+                                return false,err
+                        }
+                }
+		err := syscall.Mount("/dev/"+path, "/u-root/"+path, filesystem, flags, "")
+		if err == nil {
+			return true, nil
+		}
+	}
+	returnValue = false
+	return returnValue, nil
+}
+
+func umountEntry(path string) bool {
+	var returnValue bool
+	var flags int
+	// Was supposed to be unecessary for kernel 4.x.x
+	flags = syscall.MNT_DETACH
+	err := syscall.Unmount(path, flags)
+	if err == nil {
+		return true
+	}
+	returnValue = false
+	return returnValue
+}
+
+// checkBootEntry is looking for grub.cfg file
+// and return absolute path to it
+func checkBootEntry(mountPoint string) string {
+	_, err := os.Stat(mountPoint + "/boot")
+	if err == nil {
+		// The boot directory is there
```

Comment:
You *might* just do this:
ioutil.ReadFile("/boot/grub/grub.cfg")
this will ensure it's there and readable without having to do all these stats.

Reply:
Further, you could return both the contents of the grub.cfg and the path you found it, and then you don't have to open it in getFileMenuContent. 

This would make writing a test much easier as getFileMenuContent could, for example, take a string and you could verify the processing it does.

Reply:
Done